solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -0,0 +1,59 @@
+---
+id: ac-GCDOnBlackboard
+source: AC
+title: GCD On Blackboard
+author: Andrew Cheng
+---
+
+**Time Complexity**: $\mathcal{O}(N \log N)$ (GCD operations take $\mathcal{O}(\log N)$ time)
+
+**Main Idea:**
+The question in its essence asks us to find the maximum possible GCD of the remaining $N-1$ numbers
+after taking any one of them away. Though naively trying every single combination will result in a
+complexity of $\mathcal{O}(N^2)$ as there are $N$ different ways to take away a number and calculating
+the GCD will take $\mathcal{O}(N)$ time per case. 
+
+To speed this process up, we can utilize a prefix sum and a suffix sum. Let $l[i]$ = $GCD_{j=1}^{i} a[j]$ and $r[i]$ = $GCD_{j=i}^{N} a[j]$.
+Then the answer is the maximum of $GCD(l[i-1],r[i+1]),$ $i \in [2,N-1]$. Though remember to check for the case
+where the first or the last number is replaced. (line 25 of code)
+
+**Solution:**
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+
+#include <bits/stdc++.h>
+using namespace std;
+
+const int maxN = 1e5+5;
+
+int a[maxN];
+int l[maxN]; //l[i] = GCD of a1,a2, ..., ai
+int r[maxN]; //r[i] = GCD of ai, ai+1, ..., an
+int N;
```

Comment:
more intuitive variable names
i'd suggest `arr`, `prefGcd`, `suffGcd`, and `size` respectively
this way the names are self-documenting

Reply:
Using "size" instead of "N" will actually cause CE on the AtCoder server as "size" results in a conflict with a random system variable. I will opt for "N" instead of "size" here. The other suggestions are all adapted.